aten/src/ATen/core/jit_type.h

```diff
@@ -324,6 +331,8 @@ inline c10::optional<T> merge_primitive(
   return c10::optional<T>{};
 }
 
+enum class ContiguityType { UNKNOWN, NONCONTIGUOUS, CONTIGUOUS, BROADCASTED };
```

Comment:
i'm not sure we actually need UNKNOWN, if after a few runs we saw CONTIGUOUS -> BROADCASTED or vice versa or to NONCONTIGUOUS we can just use NONCONTUGOUS.